src/app/tests/TestSceneTable.cpp

```diff
@@ -1223,6 +1247,47 @@ TEST_F(TestSceneTable, TestHandlerFunctions)
 
         // Clear buffer
         memset(buffer, 0, buff_span.size());
+
+        LCextensionFieldSet.clusterID          = MockClusterId(kFakeClusterId);
+        LCextensionFieldSet.attributeValueList = MockFKPairs;
+
+        writer.Init(mock_LC_buffer);
+        EXPECT_EQ(CHIP_NO_ERROR, app::DataModel::Encode(writer, TLV::AnonymousTag(), LCextensionFieldSet.attributeValueList));
+        LC_buffer_serialized_length = writer.GetLengthWritten();
+
+        // Reinit buffer
+        buff_span = MutableByteSpan(buffer);
+
+        reader.Init(Mock_LC_list);
+        extensionFieldValueCapIn.clusterID = MockClusterId(kFakeClusterId);
+        EXPECT_EQ(CHIP_NO_ERROR, reader.Next());
+        EXPECT_EQ(CHIP_NO_ERROR, extensionFieldValueCapIn.attributeValueList.Decode(reader));
+
+        // Verify that the initial values are not capped
+        auto iteratorFK = extensionFieldValueCapIn.attributeValueList.begin();
+        iteratorFK.Next();
+        pair = iteratorFK.GetValue();
+        EXPECT_EQ(pair.valueUnsigned64.Value(), MockFKPairs[0].valueUnsigned64.Value());
+        iteratorFK.Next();
+        pair = iteratorFK.GetValue();
+        EXPECT_EQ(pair.valueSigned64.Value(), MockFKPairs[1].valueSigned64.Value());
+
+        // Verify that we cap the value to the mock attribute size when serializing
+        EXPECT_EQ(CHIP_NO_ERROR, mpSceneHandler->SerializeAdd(kTestEndpoint4, extensionFieldValueCapIn, buff_span));
+        EXPECT_EQ(CHIP_NO_ERROR,
+                  mpSceneHandler->Deserialize(kTestEndpoint4, MockClusterId(kFakeClusterId), buff_span, extensionFieldValueCapOut));
+
+        // Verify that the output value is capped to uint40 max value
+        uint64_t uint40Max = static_cast<uint64_t>(0x000000FFFFFFFFFF);
+        EXPECT_EQ(uint40Max, static_cast<uint64_t>(extensionFieldValueCapOut.attributeValueList[0].valueUnsigned64.Value()));
+
+        // Verify that the output value is not capped
```

Comment:
That's ... not what the code ends up testing?

Reply:
bad copy paste